src/stomach.h

```diff
@@ -85,12 +85,12 @@ class stomach_contents
         void mod_nutr( int nutr );
         // changes water amount in stomach
         // overflow draws from player thirst
-        void mod_water( units::volume h2o );
+        void mod_water( const units::volume &h2o );
```

Comment:
Once before I persuaded you to not convert `units` values to `const&` arguments (because they're just an `int`).  But if your static analysis is going to keep suggesting it we might as well follow along.  It won't really make any significant performance impact.

Reply:
I was curious what this looked like in practice because this seems like something a compiler can figure out and optimize away so I did some testing and it doesn't :(.

main.cpp
```C++
#include "a.h"
void main() {
    A a;
    int i = 5;
    int j = 7;
    a.foo(i);
    b.baz(j);
}
```
a.h
```C++
class A {
public:
    void foo( const int &bar );
    void baz( const int bar );
}
```
a.cpp
```C++
#include "a.h"
#include <stdio.h>

void A::foo( const int &bar ) {
    printf( "bar: %d\n", bar );
}
void A::baz( const int bar ) {
   printf( "bar %d\n", bar );
}
```

Then with `g++ a.h a.cpp main.cpp -o main -O3`, the calls in main look like
![image](https://user-images.githubusercontent.com/60370/59984278-273b0080-95dd-11e9-87f8-b30dbb567570.png)

Where it has to do a `movl` to put 5 into the stack for `foo`, meanwhile it gets to do a `mov` to put 7 into the `%esi` register for `baz`

Then in the actual calls themselves it has to do an extra mov from `%rsi` to `%esi` for `foo`
![image](https://user-images.githubusercontent.com/60370/59984323-acbeb080-95dd-11e9-92b6-b266c4c09d31.png)

So indeed no compiler optimizing away a const reference to an int sadly.  But an extra `mov` and two memory ops instead of a register op are probably ok at the end of the day.

Reply:
> So indeed no compiler optimizing away a const reference to an int sadly.

How could it? A reference can be converted into a pointer and that can be compared:

```C++
const int global = 27;

void func( const int &ref ) {
    if( &ref == &global ) {
        print("global");
    } else {
        print("something else");
    }
}

int main() {
    func(global_A); // "global"
    func(int(global)); // forward a temporary int, therefor "something else"
}
```

Reply:
I was theorizing it had enough information available to determine if it was safe to convert it back to an int instead of keeping it a reference but I suppose that would be a rather complex optimization.